### ISSUE TITLE
Clarify that `useAsyncData` results are not cached

### ISSUE THREAD

**Issue by gibkigonzo** [nuxt/nuxt#22622]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.16.1
- Nuxt Version:     3.6.1
- Nitro Version:    2.5.2
- Package Manager:  yarn@1.22.19
- Builder:          vite
- User Config:      runtimeConfig, alias, css, components, vite, dir, plugins, modules, pwa
- Runtime Modules:  nuxt-jsonld@2.0.8, @vite-pwa/nuxt@0.1.0, @pinia/nuxt@0.4.11, @nuxtjs/eslint-module@4.1.0
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-tex5o8?file=pages%2Findex.vue

### Describe the bug

I've checked docs and as I see `useAsyncData` should cache data for response with same key.

So for example:
```
const KEY = 'test';
const { data: data } = await useAsyncData(KEY, async () => {
  console.log('test');
  return {};
});
const { data: data2 } = await useAsyncData(KEY, async () => {
  console.log('test');
  return {};
});
const { data: data3 } = await useAsyncData(KEY, async () => {
  console.log('test');
  return {};
});
```

This should show only once `'test'` in console, but it shows it three times.

In real world app I need to fetch data in component A and then use same data in page B.
Right now the only option that I see is to make two api calls.

In nuxt 2, I've used `fetch` component option method to make call in component A and I didn't even need to make call in page B, because I've already had data in vuex (from component A). How to handle that (without duplicated requests) if `useAsyncData` doesn't provide cache?

**Comment by gibkigonzo** [comment]
As I understand docs https://nuxt.com/docs/getting-started/data-fetching#effective-caching it should be cached :thinking:

**Comment by danielroe** [comment]
`useAsyncData` does not cache results between invocations, though it does avoid refetching on client if it has been fetched on server. Cache options are passed directly to the `fetch` request, which may use the browser cache options.

A PR to clarify this would be welcome.

**Comment by gibkigonzo** [comment]
@danielroe thanks for response. Now I get it. It uses browser cache. I don't want to create new issue and I think it is somehow related. According to my real app problem:
```
In nuxt 2, I've used fetch component option method to make call in component A and I didn't even need to make call in page B, because I've already had data in vuex (from component A).
```
Here is reproduction with `useAsyncData` https://stackblitz.com/edit/github-9cq2cx?file=store%2Fmenu.ts,components%2FMenuHeader.vue,pages%2Findex.vue

So I called `useAsyncData` in one component and in other component I want to use data from `pinia`. But I can't do that because there is a missmatch (no data in page component in SSR). How to solve that problem? Should I make a call to `useAsyncData` each time when I want to get same data?

**Comment by adrienZ** [comment]
> Cache options are passed directly to the fetch request, which may use the browser cache options

are you referring to [cache control](https://developer.mozilla.org/fr/docs/Web/HTTP/Headers/Cache-Control) ?

**Comment by danielroe** [comment]
> are you referring to [cache control](https://developer.mozilla.org/fr/docs/Web/HTTP/Headers/Cache-Control) ?

Exactly.

**Comment by Zebnastien** [comment]
Hi wouldn't it be nice if Async Data could cache between invocations? 
This would be a great feature to use with Supabase, Prisma, Firebase...

I too thought that was already the case... or did it change ?

**Comment by adrienZ** [comment]
> Hi,
> 
> Wouldn't it be nice if useAsyncData could cache response between invocations? This would be a great feature to use with Supabase, Prisma, Firebase, AppWrite...
> 
> Moreover it could help reducing cost usage from services with cost based on api calls. I too thought that was already the case... or did it change ?

If I understood well, it only caches to avoid running the same request twice (as your component will be run once server-side, once client side). 

Implementing database cache is up to you, however you might too take a look at https://unstorage.unjs.io/, it comes with a lot of integrations (redis, azure, cloudflare, filesystem, etc) :)
